File: problems_001to025/problem_012/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch
from toolbox import natural_sum, positive_divisors


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = None
        num = 1
        while answer is None:
            triangle_number = natural_sum(num)
            divisors = positive_divisors(triangle_number)
            num += 1
            if len(divisors) > 500:
                answer = triangle_number
                break
        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_001to025/problem_012/solution.js
const { naturalSum, positiveDivisors } = require('../../lib/toolbox');

const solution = () => {
  let answer = null;

  let num = 1;

  while (answer === null) {
    const triangleNumber = naturalSum(num);
    const divisors = positiveDivisors(triangleNumber);

    if (divisors.length > 500) {
      answer = triangleNumber;
      break;
    }

    num += 1;
  }

  return answer;
};

if (process.env.NODE_ENV !== 'test') {
  // eslint-disable-next-line no-console
  console.log(solution());
}

module.exports = solution;
